### PR TITLE
Fix/use correct api clients for objects

### PR DESCRIPTION
**Pull request by mkrull**

### CODE REVIEW

**Comment by mkrull** [comment]
Fixes #9

File: twyla/kubedeploy/kube.py [hunk 1]
```diff
@@ -88,11 +90,55 @@ def type_name_from_data(self, data):
...
-        return api_client.deserialize(res, kubernetes_type)
+        obj = api_client.deserialize(res, kubernetes_type)
+
+        # Hackaround missing intOrString support in the python client
+        # TODO: check how to fix that upstream
+        if kubernetes_type.endswith('Deployment'):
+            try:
+                int_val = int(obj.spec.strategy.rolling_update.max_surge)
+                obj.spec.strategy.rolling_update.max_surge = int_val
+            except:
```
Comment (afroisalreadyinu) [inline twyla/kubedeploy/kube.py]: Catching every exception here does not look like a good idea. If you want to simply ignore the case where the `max_surge` attribute cannot be turned into an int, the exception you should catch is `ValueError`.

File: twyla/kubedeploy/kube.py [hunk 2]
```diff
@@ -88,11 +90,55 @@ def type_name_from_data(self, data):
...
+                pass
+            try:
+                int_val = int(obj.spec.strategy.rolling_update.max_unavailable)
+                obj.spec.strategy.rolling_update.max_unavailable = int_val
+            except:
```
Comment (afroisalreadyinu) [inline twyla/kubedeploy/kube.py]: Here too

File: twyla/kubedeploy/kube.py [hunk 3]
```diff
@@ -88,11 +90,55 @@ def type_name_from_data(self, data):
...
+                pass
+        elif kubernetes_type.endswith('Service'):
+            for port in obj.spec.ports:
+                try:
+                    int_val = int(port.target_port)
+                    port.target_port = int_val
+                except:
```
Comment (afroisalreadyinu) [inline twyla/kubedeploy/kube.py]: And here.

**Pull request merged**